packages/blade/src/components/Text/Decisions/Decisions.md

```diff
@@ -0,0 +1,87 @@
+The text component is categorised into multiple parts
+
+- BaseText - Internal Component, not published to consumers
+- Title - Public Component
+- Heading - Public Component
+- Text - Public Component
+- Code - Public Component
+
+Here's how the hierarchy will looks like
+![Component Breakdown](./component-breakdown.png)
+
+The APIs for all the above component will look something like below
+
+- `BaseText`
+
+  This component is just meant to be used for internal purposes for Blade
+  Here's how the API will look like
+
+  ```jsx
+  <BaseText
+    color="surface.text | feedback.text | button.text"
+    fontFamily="fontfamily.name"
+    fontSize="fontSize.name"
+    fontWeight="fontWeight.name"
+    lineWeight="lineWeight"
+    letterSpacing="letterSpacing"
+    formatting="bold/italic/strike"
+    as="h1-h6/p/span"
+  />
+  ```
+
+- `Title`
+
+  This component is meant to be used for Titles(bigger banner style kind of text) mostly on landing pages or page level headings.
+
+  Title will internally use `BaseText` to render. We'll use native html heading tags to render title and native text element for react-native to render the Title.
+
+  ```jsx
+  <Title variant="small | medium | large" type="normal | subtle | subdued | muted | placeholder">
+    Some Title
+  </Title>
+  ```
+
+- `Heading`
+
+  As the name indicates these will be used for headings on the page.
+
+  Heading will internally use `BaseText` to render. Natively we'll use html heading tags for web and native text element for react-native to render the headings.
+
+  ```jsx
+  <Heading
+    variant="small | medium | large | subheading"
+    type="normal | subtle | subdued | muted | placeholder"
+  >
+    Some Heading
+  </Heading>
+  ```
+
+- `Text`
+
+  This component is the one that we'll use to render normal text on the screen(a.k.a as paragraphs or body text)
+
+  Text will internally use `BaseText` to render. Natively we'll use html `<p></p>` tag for web and native `<Text></Text>` react-native to render the headings.
+
+  ```jsx
+  <Text
+    variant=" medium | large | caption"
+    weight="regular | bold"
+    type="normal | subtle | subdued | muted | placeholder"
+    shouldTruncate="true | false"
```

Comment:
We might need to be consistent of these naming conventions where we prefix booleans with "should" or "is". 

Feel like isTruncated is a better here. 🤔

Reply:
Should we just call it `truncate`? 

`truncate` sounds more like a command we're giving to the component and makes `true` seem implicit. Since the default is `false`, the presence or absence of `truncate` will act as a decision-maker. `shouldTruncate` sounds more like a question that needs an answer - should truncate? yes.  Also, `truncate` is much smaller to write 😛 

```jsx
<Text
  truncate
/>
```

Reply:
truncate is fine too, although i will lean a bit towards prefixing with `is`. 

reason being: 

- prefixing with `is` hints user that its a boolean without knowing the types.
- `truncate` can be a bit confusing in non-TS projects like: "What does it take? truncate={5} will this truncate 5lines? or what?"
- Seeing most third party design systems also follow this naming convention

Reply:
Makes sense. `is` as a prefix makes more sense to me than `should`. For instance, a button can have `disabled` or `isDisabled` but never `shouldDisable`

Reply:
there's a difference in tense when you refer to `is` and `should`. `is` basically says it will instantly be true/false or on action for eg: `isDisabled` for a button that might be toggled based on the form being submitted or based on a user role. when you talk about `should` it basically says the behavior shall be toggled for something that might happen in the future eg: `shouldComponentUpdate` lifecycle method. Similarly, over here we are saying `shouldTruncate` which dictates that the text shall be truncated after some lines so if I say `isTruncate` that conveys that it'll immediately be truncated but ideally it's not.